### PR TITLE
More Gateway Stuff!

### PR DESCRIPTION
**Pull request by Very-Soft**
Some new gateway things!

image
New dirt

denecrotizer-export
Adds the denecrotizer, an item for reviving dead simplemobs. The experimental version also changes the mob's faction to yours, and offers them up to ghosts to play as. (The experimental version also works on still-alive passive mobs, so you don't have to kill Ian to invite a ghost to play as him.) The things that can be revived have been somewhat restricted to prevent abuse. (No nurse spiders, no carrier spiders, no mechs/robots/mercs, etc.)

m denecrotizer-export

Medical gets a version of this too! It follows the same rules for reviving, but doesn't change the mob's faction or offer it to ghosts! This one would go into medical's arsenal, to revive dead pets or player controlled mobs. (It can be used on hostile things too, but it doesn't make them not hostile.) ((This version has a shorter cooldown and more charges than the experimental version))

image
There's also new sidewalks and trees now!

Adds new maps for the gateway. Some dangerous, others not so much. Some of these maps utilize a new kind of away gateway, which requires players to locate an item to use on the gateway to activate it for returning.

### CODE REVIEW

**Comment by dwinters99** [comment]
This is very nice, however depending on @Novacat's plan for the gateway you may want to leave it commented out in tether_defines.dm until the gateway is re-added officially.

File: vorestation.dme [hunk 1]
```diff
@@ -1164,6 +1165,7 @@
 #include "code\game\objects\items\devices\chameleonproj.dm"
 #include "code\game\objects\items\devices\debugger.dm"
 #include "code\game\objects\items\devices\defib.dm"
+#include "code\game\objects\items\devices\denecrotizer.dm"
```
Comment (Novacat) [inline vorestation.dme:1168]: This should probably have _vr at the end so we know it is a vorestation-specific file.

**Review by Novacat — CHANGES_REQUESTED** [review]

**Review by Novacat — APPROVED** [review]

**Comment by Ketrai** [comment]
Hey soft, could you look into using smooth shading instead of noise for the sidewalk tiles? Not saying noise is bad in general, but it can make tiles look pretty busy. Or alternatively, blend in the noise with gradients, edges, lesser contrast.

image
Starbound, most tiles use smooth or limited shading with connecting textures. Earlier tiles use more noise, tiles that were made later don't use noise, or use it sparingly with rather soft shades that blend in better. 
image
versus
image
Second tile still uses noise, but it's much more muted. You don't even really notice it's there ingame, which I feel is the perfect implementation.
https://cdn.discordapp.com/attachments/404984611551182849/824401056695451658/2021-03-24_16-53-25.png (Link instead of image due to file size.)
In vintage story, virtually all blocks use noise texturing. Whilst it looks muddied up close as can be seen in the screenshot below, from afar, you don't see that much of it. They also use a bunch of gradients to blend in the noise more effectively for plants, aka objects you're more likely to examine up close.
image
Finally some of my own experiments with texturing. Mostly gradients, but the wood tiles to the right in particular use noise on the edges. And the older wood tiles use noise through-and-through. 
image

**Comment by Ketrai** [comment]
Oh, one of my favorite blocks wasn't included in that tile comparison screenshot. 
image
Concrete here also uses noise in a great way here. 3 tones that don't contrast harshly. With sharp edges and highlights to set apart the edges even more, whilst making the walls pop forwards a bit. The noise is also structured in clusters, like in your dirt tile. Or in other words, most of the time a differently colored pixel will be surrounded by a couple pixels of its own shade. Rather than being surrounded entirely by a different shade, drawing attention.

**Comment by Ketrai** [comment]
image
Quick brush-over just to show what it could look like. I've applied a highlight. Deepened the contrast in the gaps between tiles. And smoothened over the noise to be more subtle. I didn't use clusters yet, because that would be quite a bit of extra work. (at least I don't know a way to do them quick other than hand-painting, then copy-pasting some of the texture work and mixing it)

**Comment by Very-Soft** [comment]
> image
> Quick brush-over just to show what it could look like. I've applied a highlight. Deepened the contrast in the gaps between tiles. And smoothened over the noise to be more subtle. I didn't use clusters yet, because that would be quite a bit of extra work. (at least I don't know a way to do them quick other than hand-painting, then copy-pasting some of the texture work and mixing it)
> 
> Edit : updated the screenshot as I forgot a lil hard edge at the bottom of each tile.

I don't disagree with your approach here, but I wasn't going for a shiny well maintained or interior tile, I was going for a weathered concrete exterior tile that would go well with grass and dirt.  

Your one looks nice and shiny, smooth and pristine, like someone cares for it regularly, where mine looks like you would scrape your face off if you fell on it. I tried to sell that too with the fact that there are a bunch of alternative sprites with cracks and grass growing up out of it. (I just kind of got unlucky in the screenshot)

**Comment by Ketrai** [comment]
I've seen the variants with cracks, and I always like variation. However, I don't feel like this is the right approach to take to make a tile look weathered. 
image
image

From afar, the tile I've made doesn't look much shinier at all, always compare both what a tile looks like up close and in the actual game environment, aka being rendered at 32x32. The highlights and edges also aren't just about making it look more smooth, it's about creating an illusion of 3D.

**Comment by Very-Soft** [comment]
> I've seen the variants with cracks, and I always like variation. However, I don't feel like this is the right approach to take to make a tile look weathered.
> image
> image
> 
> From afar, the tile I've made doesn't look much shinier at all, always compare both what a tile looks like up close and in the actual game environment, aka being rendered at 32x32. The highlights and edges also aren't just about making it look more smooth, it's about creating an illusion of 3D.

I should also note that there's kind of a difference between designing sprites like this, and for something like starbound as your examples show. I don't disagree with what you've said, but in starbound things are viewed side on, you can always assume that light comes from north and shade and highlight things accordingly. 

I think that like, walls generally are best when they follow that principle, or other things that are supposed to stand vertically. But for things that are flat on the ground or otherwise horizontally oriented toward the view, such highlighting implies shine and edges with depth. 

I think you are focused on the depth, and not on other creature factors, such as easily noticeable repeating patterns and smoothness.

Art is also subjective. To me, your example does have more of a 3 dimensional feel, but it also looks shiny, and the shiny pattern is highly repetitive and easy to notice when you put them in a line. The noisier pattern and more similar but obviously different coloring for the edges doesn't sell the depth as well, but it's easy on the eyes and when I look at it in a line I don't see much in the way of repeating patterns outside of the ones that are intended.

**Comment by Ketrai** [comment]
image

Left is an example with modest use of sharp noise to create the feeling of a weather, more damaged tile. Right is just keeping the shinier texturing whilst making the tile overall darker.

**Comment by Very-Soft** [comment]
image

Like, I can walk you through my design process with the thing. The things I was going for with it~ 

First, I designed it with the lines horizontally oriented, and just started with some grey, played around with gradients and such, and I kept being unsatisfied with how it looked for a variety of reasons. I thought about real sidewalks though, and how a lot of times they are rough and a bit icky, with a lot of times there being a sort of more worn in or lighter portion at the center, where people have most often walked/biked/whatever. 

So I approached it from that angle. It only really has noticably repeating patterns in
-the lines denoting the edges
-the lighter coloring along the center of the path.

**Comment by Ketrai** [comment]
image
Another one I like is cutting ribbons, or making a patch. The thing is with your approach is that it feels random, which is good... but it also feels too random, like there isn't a clear texture anymore. In this example the ribbons layer an consistency in, namely that you've drawn lines. Whilst a patch will segment the texture into two noticeable areas. Note that these are incomplete examples, you'll want to add some minor noise to the rest of the space too. Just less so there's a contrast between your patch/ribbons and the rest of the tile. These can feel JUST as random too, just random in a way nature intended it.

**Comment by Ketrai** [comment]
https://www.ksmu.org/sites/ksmumain/files/styles/x_large/public/201712/IMG_1256.JPG
Very noisy sidewalk. Except the textures blend together more than in pixelart because there's well, thousands of pixels so to speak. You can also see "ribbons" of different texture around the edges, and a slightly gradual shadow at the gaps rather than a sharp cut. 

https://bloximages.newyork1.vip.townnews.com/yakimaherald.com/content/tncms/assets/v3/editorial/4/1d/41dd19bd-f8f0-567e-8263-dddf3559d811/5d1268deba897.image.jpg?crop=1175%2C1175%2C294%2C0&resize=1200%2C1200&order=crop%2Cresize
Here's a sidewalk that has very clear ribbons, and additional weathered patches around the sides.

**Comment by Ketrai** [comment]
image
More refined version to mimic the second screenshot from above, minus a few random specks of dirt.

**Comment by Very-Soft** [comment]
I think we're going to have to agree to disagree here dood. You're welcome to make your own alternative sidewalk sprite. I think your sprites are fine, but they aren't really what I was going for. I can understand that you might not see what I was going for, and that's fine, but I'm not going to redo the sprite in your style.

**Comment by Ketrai** [comment]
image
image
More alternatives, untextured ribbons, and strongly hilighted ribbons.

**Comment by Ketrai** [comment]
https://cdn.discordapp.com/attachments/308880769965883392/853939383139565588/unknown.png
Splotches of gradients can also work to make it look less busy.

**Comment by Very-Soft** [comment]
> image
> Splotches of gradients can also work to make it look less busy.

Easily noticeable patterns. I appreciate your advice, I am not changing the sprite based on it.

**Comment by Ketrai** [comment]
Welp, then all else I can say is that most modern gray sidewalks look quite bland and I wish for a more colorful version or pattern if I'm to use these in any mapping projects I do.

**Comment by Very-Soft** [comment]
> Welp, then all else I can say is that most modern gray sidewalks look quite bland and I wish for a more colorful version or pattern if I'm to use these in any mapping projects I do.

As I said, you are welcome to make your own variant of sidewalk, or as many variants, in however many colors or patterns you like. I made this one for this gateway project. I imagine other people will make other turfs for their projects too. I don't disagree with your ideas. They just aren't what I was going for with regards to this project.

**Review by Arokha — APPROVED** [review]

**Pull request merged**